### PR TITLE
Define types of items in $this->global in compiled template

### PR DESCRIPTION
**Pull request by MartinMystikJonas**

### CODE REVIEW

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php [hunk 1]
```diff
@@ -27,6 +27,26 @@ public function testVariables(): void
                 5,
                 'default.latte',
             ],
+            [
```
Comment (lulco) [inline tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:30]: I would like to not report issues which user/developer can't fix from latte. These are some internal nette/latte errors right? We should fix them by modifying compiled template or ignore them.

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php [hunk 2]
```diff
@@ -143,6 +163,11 @@ public function testComponents(): void
     public function testFilters(): void
     {
         $this->analyse([__DIR__ . '/Fixtures/FiltersPresenter.php', __DIR__ . '/Fixtures/ParentPresenter.php'], [
+            [
+                'Trying to invoke mixed but it\'s not a callable.',
```
Comment (lulco) [inline tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:167]: this is covered by message Undefined latte filter, imho we can ignore it

File: src/VariableCollector/DefaultTemplateVariables.php [hunk 3]
```diff
@@ -21,6 +21,7 @@ public function collect(): array
         $variables[] = new Variable('baseUrl', new StringType());
         $variables[] = new Variable('basePath', new StringType());
         $variables[] = new Variable('ʟ_fi', new ObjectType('Latte\Runtime\FilterInfo'));
+        $variables[] = new Variable('ʟ_tag', new ArrayType(new MixedType(), new MixedType()));
```
Comment (lulco) [inline src/VariableCollector/DefaultTemplateVariables.php]: don't we know the exact or more strict type of this variable?

**Review by lulco — COMMENTED** [review]
There are few errors in unit tests, check them please

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php (hunk 2 above)
Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:167]: In this case yes but same error can occur in different context caused by user code error

File: src/VariableCollector/DefaultTemplateVariables.php (hunk 3 above)
Comment (MartinMystikJonas) [reply src/VariableCollector/DefaultTemplateVariables.php]: I think it is string[] but not quite sure without closet examination of latte internals

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php (hunk 1 above)
Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:30]: It is not latte internal but secondary errors caused by use of unknown (mixed) variable. It is imho expected to occur on level 9. Same errors would occur in php code when foreach iterates on mixed and you try access properties of mixed.

**Comment by MartinMystikJonas** [comment]
I will check tests errors - weirs that it did not occured in local. Maybe different version of some dependency.

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php (hunk 1 above)
Comment (lulco) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:30]: ok then

File: tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php (hunk 2 above)
Comment (MartinMystikJonas) [reply tests/Rule/LatteTemplatesRule/PresenterWithoutModule/LatteTemplatesRuleForPresenterTest.php:167]: For example when user passes callable in variable and tries to invoke it in template then same error will occur. So we cannot ignore it. Only option would be conditional ignore I mwntioned - ignore this error only if undefined filter error is present on same location

File: src/VariableCollector/DefaultTemplateVariables.php (hunk 3 above)
Comment (MartinMystikJonas) [reply src/VariableCollector/DefaultTemplateVariables.php]: It seems to be string[] I will narrow it down.

**Review by lulco — APPROVED** [review]

**Pull request merged**